src/workerd/api/sockets.c++

```diff
@@ -139,6 +139,8 @@ jsg::Ref<Socket> connectImplNoOutputLock(
 
   auto& ioContext = IoContext::current();
 
+  JSG_REQUIRE(!ioContext.isFiddle(), TypeError, "Socket API not supported in fiddle mode.");
```

Comment:
I think end-users are unlikely to know what "fiddle mode" means.  Might be more clear to say something like "web preview"?

Also not sure if this is covered by existing tests, but if not, would be good to have coverage for it eventually?

Reply:
Ditto